sdk/src/main/scala/io/horizen/history/AbstractHistory.scala

```diff
@@ -600,9 +601,36 @@ abstract class AbstractHistory[
       case None => Optional.empty()
     }
   }
+
+  def tooManyBlocksWithoutMcRefs(parentBlockId: ModifierId, noMcRefInCurrentBlock: Boolean, consensusEpochNumber: Int): Boolean = {
+    if (noMcRefInCurrentBlock) {
+      if (!Version1_3_0Fork.get(consensusEpochNumber).active) {
+        // fork is not active for computing the number of sc blocks without mc refs
+        log.debug(s"Fork for version 1.3 not active, skipping counting of sc blocks without mc references")
+        false
+      } else {
+
+        val count = storage.getNumOfScBlocksWithNoMcRefDataSince(parentBlockId) + 1
+        if (count >= params.maxHistoryRewritingLength) {
+          log.warn(s"####### Num of SC blocks with no mc refs: $count >= max number of revertible sc blocks (${params.maxHistoryRewritingLength})")
+          true
+        } else {
+          // they are not too many, since we are below the critical threshold
+          log.debug(s"Num of SC blocks with no mc refs: $count")
```

Comment:
also here I would remove the debug message

Reply:
Ok, done